src/qibo/backends/_clifford_operations.py

```diff
@@ -329,7 +329,7 @@ def ECR(symplectic_matrix, control_q, target_q, nqubits):
     return X(symplectic_matrix, control_q, nqubits)
 
 
-def _exponent(x1, z1, x2, z2):
+def _exponent(x1: np.array, z1: np.array, x2: np.array, z2: np.array) -> np.array:
```

Comment:
In case you care, `np.array` is a function, so it's not suitable as a type hint.

The correct type would be `np.ndarray` (just try `type(np.array([]))`), but an even better typing support is coming from the `numpy.typing` module:
https://numpy.org/doc/stable/reference/typing.html